lib/logstash/codecs/multiline.rb

```diff
@@ -297,4 +297,9 @@ def auto_flush_active?
   def auto_flush_runner
     @auto_flush_runner || AutoFlushUnset.new(nil, nil)
   end
+
+  def initialize_copy(source)
+    super
+    register
```

Comment:
`LogStash:Codecs::Multiline#register` does things like hitting the filesystem to load grok patterns, recompiling the pattern, etc.; do we need to do all of this each time we flush?

Reply:
@yaauie this is not called when flushing, this is only called when setting up a new instance of the codec via `.dup` or `.clone` as we do in all the multi-threaded input plugins.

Reply:
Gotcha. In that case. LGTM.